scripts/quests/outlands/NIN_AF3_True_Will.lua

```diff
@@ -0,0 +1,176 @@
+                        quest:getVar(player, 'nmKilled') == 0
+                    then
+                        quest:setVar(player, 'nmKilled', 1)
+                    end
+                end,
+            },
+
+            ['Kappa_Biwa'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if
+                        GetMobByID(ID.mob.KAPPA_AKUSO):isDead() and
+                        GetMobByID(ID.mob.KAPPA_BONZE):isDead() and
+                        quest:getVar(player, 'nmKilled') == 0
+                    then
+                        quest:setVar(player, 'nmKilled', 1)
+                    end
+                end,
+            },
+
+            ['Kappa_Bonze'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if
+                        GetMobByID(ID.mob.KAPPA_BIWA):isDead() and
+                        GetMobByID(ID.mob.KAPPA_AKUSO):isDead() and
+                        quest:getVar(player, 'nmKilled') == 0
+                    then
+                        quest:setVar(player, 'nmKilled', 1)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.NORG] =
+        {
+            ['Ryoma'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        player:hasKeyItem(xi.ki.OLD_TRICK_BOX) and
+                        quest:getVar(player, 'Prog') == 0
+                    then
+                        return quest:progressEvent(137)
+                    elseif quest:getVar(player, 'Prog') == 1 then
+                        return quest:event(138)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [137] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.RABAO] =
+        {
+            ['Leodarion'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(97)
+                    elseif
+                        quest:getVar(player, 'Prog') == 2 and
+                        not player:hasKeyItem(xi.ki.LARGE_TRICK_BOX)
+                    then
+                        return quest:event(98)
+                    elseif player:hasKeyItem(xi.ki.LARGE_TRICK_BOX) then
+                        return quest:progressEvent(99)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [97] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                    player:delKeyItem(xi.ki.OLD_TRICK_BOX)
+                    return quest:messageSpecial(rabaoID.text.KEYITEM_LOST, xi.ki.OLD_TRICK_BOX)
```

Comment:
We're already returned quest:progressEvent(xx)
We can't return another entity, so here you can just use player:messageSpecial